File: NetFramework/WinForms/CS/ControlExplorerOld/FlexGrid/CustomFilters/StringFilterEditor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace ControlExplorer.FlexGrid.CustomFilters
{
    /// <summary>
    /// Represents an editor for modifying a <see cref="StringFilterEditor"/>.
    /// </summary>
    public partial class StringFilterEditor : 
        UserControl,
        C1.Win.C1FlexGrid.IC1ColumnFilterEditor
    {
        //-------------------------------------------------------------------------------
        #region ** fields

        StringFilter _filter;

        #endregion

        //-------------------------------------------------------------------------------
        #region ** ctor

        public StringFilterEditor()
        {
            InitializeComponent();
        }

        #endregion

        //-------------------------------------------------------------------------------
        #region ** IC1ColumnFilterEditor

        public void Initialize(C1.Win.C1FlexGrid.C1FlexGridBase grid, int columnIndex, C1.Win.C1FlexGrid.IC1ColumnFilter filter)
        {
            _filter = (StringFilter)filter;

            // initialize checkbox values
            foreach (var pt in _filter.Ranges)
            {
                switch ((char)pt.X)
                {
                    case 'A':
                        _chkAE.Checked = true;
                        break;
                    case 'F':
                        _chkFJ.Checked = true;
                        break;
                    case 'K':
                        _chkKO.Checked = true;
                        break;
                    case 'P':
                        _chkPT.Checked = true;
                        break;
                    case 'U':
                        _chkUZ.Checked = true;
                        break;
                }
            }
        }
        public void ApplyChanges()
        {
            // reset filter
            _filter.Ranges.Clear();

            // add selected ranges
            foreach (Control ctl in this.Controls)
            {
                var cb = ctl as CheckBox;
                if (cb != null && cb.Checked)
                {
                    var pt = new Point((int)cb.Text[0], (int)cb.Text[cb.Text.Length - 1]);
                    _filter.Ranges.Add(pt);
                }
            }
        }
        public bool KeepFormOpen
        {
            get { return false; }
        }

        #endregion

        //-------------------------------------------------------------------------------
        #region ** event handlers

        void _chkAE_CheckedChanged(object sender, EventArgs e)
        {
            var cb = sender as CheckBox;
            cb.Font = new Font(Font, cb.Checked ? FontStyle.Bold : FontStyle.Regular);
        }

        #endregion
    }
}

File: NetFramework/FlexGrid/VB/CustomFilters/StringFilterEditor.vb
Imports System.ComponentModel
Imports System.Text

Namespace CustomFilters
	''' <summary>
	''' Represents an editor for modifying a <see cref="StringFilterEditor"/>.
	''' </summary>
	Partial Public Class StringFilterEditor
		Inherits UserControl
		Implements C1.Win.FlexGrid.IC1ColumnFilterEditor
		'-------------------------------------------------------------------------------
		#Region "** fields"

		Private _filter As StringFilter

		#End Region

		'-------------------------------------------------------------------------------
		#Region "** ctor"

		Public Sub New()
			InitializeComponent()
		End Sub

		#End Region

		'-------------------------------------------------------------------------------
		#Region "** IC1ColumnFilterEditor"

		Public Sub Initialize(ByVal grid As C1.Win.FlexGrid.C1FlexGridBase, ByVal columnIndex As Integer, ByVal filter As C1.Win.FlexGrid.IC1ColumnFilter) Implements C1.Win.FlexGrid.IC1ColumnFilterEditor.Initialize
			_filter = CType(filter, StringFilter)

			' initialize checkbox values
			For Each pt In _filter.Ranges
                Select Case CChar(ChrW(pt.X))
                    Case "A"c
                        _chkAE.Checked = True
                    Case "F"c
                        _chkFJ.Checked = True
                    Case "K"c
                        _chkKO.Checked = True
                    Case "P"c
                        _chkPT.Checked = True
                    Case "U"c
                        _chkUZ.Checked = True
                End Select
			Next pt
		End Sub
		Public Sub ApplyChanges() Implements C1.Win.FlexGrid.IC1ColumnFilterEditor.ApplyChanges
			' reset filter
			_filter.Ranges.Clear()

			' add selected ranges
			For Each ctl As Control In Me.Controls
				Dim cb = TryCast(ctl, CheckBox)
				If cb IsNot Nothing AndAlso cb.Checked Then
                    Dim pt = New Point(Convert.ToInt32(cb.Text(0)), Convert.ToInt32(cb.Text(cb.Text.Length - 1)))
                    _filter.Ranges.Add(pt)
				End If
			Next ctl
		End Sub
		Public ReadOnly Property KeepFormOpen() As Boolean Implements C1.Win.FlexGrid.IC1ColumnFilterEditor.KeepFormOpen
			Get
				Return False
			End Get
		End Property

		#End Region

		'-------------------------------------------------------------------------------
		#Region "** event handlers"

		Private Sub _chkAE_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles _chkAE.CheckedChanged, _chkFJ.CheckedChanged, _chkKO.CheckedChanged, _chkPT.CheckedChanged, _chkUZ.CheckedChanged
			Dim cb = TryCast(sender, CheckBox)
			cb.Font = New Font(Font,If(cb.Checked, FontStyle.Bold, FontStyle.Regular))
		End Sub

		#End Region
	End Class
End Namespace
